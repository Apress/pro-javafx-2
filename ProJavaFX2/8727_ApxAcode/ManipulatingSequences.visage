/*
 * @test
 * @run
 */

var integers = [1, 3, 5, 7, 9];

print("integers = "); println(integers);
print("sizeof integers = "); println(sizeof integers);
print("integers[0] = "); println(integers[0]);
print("integers[4] = "); println(integers[4]);
print("integers[-1] = "); println(integers[-1]);
print("integers[5] = "); println(integers[5]);

print("integers[0..2] = "); println(integers[0..2]);
print("integers[0..<2] = "); println(integers[0..<2]);
print("integers[2..] = "); println(integers[2..]);
print("integers[-3..10] = "); println(integers[-3..10]);

print("integers[2..<2] = "); println(integers[2..<2]);
print("integers[2..0] = "); println(integers[2..0]);
print("integers[-2..-1] = "); println(integers[-2..-1]);
print("integers[5..6] = "); println(integers[5..6]);

print("integers[x | x > 4] = "); println(integers[x | x > 4]);
print("integers[x | indexof x < 2] = "); println(integers[x | indexof x < 2]);
print("integers[x | x > 10] = "); println(integers[x | x > 10]);

print("reverse integers = "); println(reverse integers);

var numbers = [3.14159, 2.71828];
print("numbers = "); println(numbers);

insert 0.57722 into numbers;
print("numbers = "); println(numbers);

insert 1.618 before numbers[0];
print("numbers = "); println(numbers);

insert 1.4142 after numbers[2];
print("numbers = "); println(numbers);

var strings = ["hello", "hello again", "goodbye"];
print("strings = "); println(strings);

insert ["how are you", "see you"] after strings[1];
print("strings = "); println(strings);

delete "see you" from strings;
print("strings = "); println(strings);

delete strings[2];
print("strings = "); println(strings);

delete strings[0..1];
print("strings = "); println(strings);

delete strings;
print("strings = "); println(strings);
