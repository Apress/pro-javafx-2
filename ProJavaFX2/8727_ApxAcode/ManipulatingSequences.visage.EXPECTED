integers = [ 1, 3, 5, 7, 9 ]
sizeof integers = 5
integers[0] = 1
integers[4] = 9
integers[-1] = 0
integers[5] = 0
integers[0..2] = [ 1, 3, 5 ]
integers[0..<2] = [ 1, 3 ]
integers[2..] = [ 5, 7, 9 ]
integers[-3..10] = [ 1, 3, 5, 7, 9 ]
integers[2..<2] = [ ]
integers[2..0] = [ ]
integers[-2..-1] = [ ]
integers[5..6] = [ ]
integers[x | x > 4] = [ 5, 7, 9 ]
integers[x | indexof x < 2] = [ 1, 3 ]
integers[x | x > 10] = [ ]
reverse integers = [ 9, 7, 5, 3, 1 ]
numbers = [ 3.14159, 2.71828 ]
numbers = [ 3.14159, 2.71828, 0.57722 ]
numbers = [ 1.618, 3.14159, 2.71828, 0.57722 ]
numbers = [ 1.618, 3.14159, 2.71828, 1.4142, 0.57722 ]
strings = [ hello, hello again, goodbye ]
strings = [ hello, hello again, how are you, see you, goodbye ]
strings = [ hello, hello again, how are you, goodbye ]
strings = [ hello, hello again, goodbye ]
strings = [ goodbye ]
strings = [ ]
